hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/listener/ClientListenerServiceImpl.java

```diff
@@ -339,7 +339,7 @@ private Boolean deregisterListenerInternal(@Nullable UUID userRegistrationId) {
                             + " has failed for address " + subscriber.getRemoteAddress(), throwable);
                 }
                 return null;
-            });
+            }).joinInternal();
```

Comment:
Would it be better to send all de-registration requests and wait for all of them to finish, using https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#allOf-java.util.concurrent.CompletableFuture...- ?

With serial execution like this, if one of the invocations fails, we will not continue sending deregistration requests to other members. 

Also, it would be faster than the serial approach.